fuse/data/visualizer/visualizer_base.py

```diff
@@ -25,17 +34,56 @@ def _show(self, vis_data: List):
         :param vis_data: preprocessed visualization items to display
         """
         raise "should implement abstract method"
+    
+    @abstractmethod
+    def _save(self, vis_data: List , output_path : str , file_name : str):
+        """
+        actual visualization function, gets a preprocessed collection of items to visualize/compare and saves the visualization to file
+        should be overriden by a specific visualizer
+        
+        :param vis_data: preprocessed visualization items to display
+        :param output_path : output folder location
+        :param file_name : file name
+        """
+        raise "should implement abstract method"
+    def _preprocess(self, vis_data: Union[List, NDict]) -> Tuple[List[Dict] , str]:
+        if isinstance(vis_data, NDict):
+            vis_data = [vis_data]
+        res = []
+        for element in vis_data :
+            vitem = TypedElement()
+            for item, value in element.items() :
+                if value['type'] is DataTypeImaging.IMAGE:
```

Comment:
Where do we set the type and how?

Reply:
it's when you give namespace as pipeline key.
example.img
example.seg
etc'

Reply:
Is it a must (the img suffix) if we want to use this visualizer?

Reply:
I think this is part of the design You did before, 
otherwise why do we need TypedElement?